portal/src/main/scala/cromwell/pipeline/service/UserService.scala

```diff
@@ -0,0 +1,34 @@
+package cromwell.pipeline.service
+
+import cromwell.pipeline.datastorage.dao.repository.UserRepository
+import cromwell.pipeline.datastorage.dto.user.{ PasswordUpdateRequest, UserUpdateRequest }
+import cromwell.pipeline.datastorage.dto.{ User, UserId }
+import cromwell.pipeline.utils.StringUtils
+import play.api.libs.json.JsError
+import play.api.libs.json.JsResult.Exception
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.Random
+
+class UserService(userRepository: UserRepository)(implicit executionContext: ExecutionContext) {
+
+  def updateUser(id: String, request: UserUpdateRequest): Future[Int] = {
+    val updatedUser =
+      User(UserId(id), email = request.email, firstName = request.firstName, lastName = request.lastName)
+    userRepository.updateUser(updatedUser)
+  }
+
+  def updateUserPassword(id: String, request: PasswordUpdateRequest): Future[Int] =
+    if (request.newPassword != request.repeatPassword) {
+      Future.failed(Exception(JsError("Password doesn't match")))
```

Comment:
Do not use `json` in the `Service` layer. The existence of the `json` in our application is an implementation detail of the `Controllers`, `Service` layer should not know about it.